app/services/katello/pulp3/repository.rb

```diff
@@ -205,7 +205,9 @@ def get_distribution(href = distribution_reference.href)
       def distribution_needs_update?
         if distribution_reference
           expected = secure_distribution_options(relative_path).except(:name).compact
-          actual = get_distribution.to_hash
+          actual = {}
+          dist = get_distribution
+          actual = dist.to_hash unless dist.nil?
```

Comment:
could you do:
```
actual = get_distribution&.to_hash || nil 
```

Reply:
Actually, the reason I didn't do this was that `actual` being `nil` would break the call to `slice`.

Reply:
oh, then why not?  
```
actual = get_distribution&.to_hash || {}
```

Reply:
Yeah that works :)